src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -102,8 +102,9 @@ public RubyContext(RubyLanguage language, TruffleLanguage.Env env) {
         this.env = env;
 
         final OptionsBuilder optionsBuilder = new OptionsBuilder();
-        optionsBuilder.set(System.getProperties());
+        optionsBuilder.set(env.getOptions());
         optionsBuilder.set(env.getConfig());
+        optionsBuilder.set(System.getProperties());
```

Comment:
Should these be in reverse order since the `set(String name, Object value)` overwrite each other?
In that case, please add a comment and a test for it.